From ba3a06edc87cd67ca8a73c08223ece3396b02298 Mon Sep 17 00:00:00 2001
From: KJ Tsanaktsidis <kj@kjtsanaktsidis.id.au>
Date: Sat, 19 Oct 2024 20:44:31 +1100
Subject: [PATCH] Add a new ASAN exclusion range

Since https://github.com/llvm/llvm-project/commit/fb77ca05ffb4f8e666878f2f6718a9fb4d686839
they actually moved the region that ASAN uses for its allocator.
---
 src/RecordSession.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/RecordSession.cc b/src/RecordSession.cc
index f370b9ab..9dee7197 100644
--- a/src/RecordSession.cc
+++ b/src/RecordSession.cc
@@ -2259,6 +2259,10 @@ static const MemoryRange asan_shadow(remote_ptr<void>((uintptr_t)0x00007fff7000L
                                      remote_ptr<void>((uintptr_t)0x10007fff8000LL));
 static const MemoryRange asan_allocator_reserved(remote_ptr<void>((uintptr_t)0x600000000000LL),
                                                  remote_ptr<void>((uintptr_t)0x640000002000LL));
+// See https://github.com/llvm/llvm-project/commit/fb77ca05ffb4f8e666878f2f6718a9fb4d686839
+// and https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/asan/asan_allocator.h
+static const MemoryRange asan_allocator_reserved2(remote_ptr<void>((uintptr_t)0x500000000000LL),
+                                                  remote_ptr<void>((uintptr_t)0x540000000000LL));
 
 // See https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/tsan/rtl/tsan_platform_posix.cpp
 static const MemoryRange tsan_shadow(remote_ptr<void>((uintptr_t)0x008000000000LL),
@@ -2287,6 +2291,7 @@ struct ExeInfo {
     }
     sanitizer_exclude_memory_ranges.push_back(asan_shadow);
     sanitizer_exclude_memory_ranges.push_back(asan_allocator_reserved);
+    sanitizer_exclude_memory_ranges.push_back(asan_allocator_reserved2);
   }
   void setup_tsan_memory_ranges() {
     if (!check_sanitizer_arch()) {
-- 
2.47.0

